packages/world/src/systems/FarmingSystem.sol

```diff
@@ -4,43 +4,41 @@ pragma solidity >=0.8.24;
 import { System } from "@latticexyz/world/src/System.sol";
 import { FixedPointMathLib } from "solady/utils/FixedPointMathLib.sol";
 
-import { ObjectType } from "../codegen/tables/ObjectType.sol";
+import { EntityObjectType } from "../codegen/tables/EntityObjectType.sol";
+import { ObjectTypeMetadata } from "../codegen/tables/ObjectTypeMetadata.sol";
 
 import { ResourceCount } from "../codegen/tables/ResourceCount.sol";
 
-import { transferEnergyToPool } from "../utils/EnergyUtils.sol";
-import { getObjectTypeIdAt, getOrCreateEntityAt } from "../utils/EntityUtils.sol";
+import { addEnergyToLocalPool, transferEnergyToPool } from "../utils/EnergyUtils.sol";
+import { getObjectTypeAt, getOrCreateEntityAt } from "../utils/EntityUtils.sol";
 import { InventoryUtils } from "../utils/InventoryUtils.sol";
 
 import { PlayerUtils } from "../utils/PlayerUtils.sol";
 
 import { TILL_ENERGY_COST } from "../Constants.sol";
 import { EntityId } from "../EntityId.sol";
-import { ObjectTypeId } from "../ObjectTypeId.sol";
-import { ObjectTypeLib } from "../ObjectTypeLib.sol";
-import { ObjectTypes } from "../ObjectTypes.sol";
+import { ObjectType } from "../ObjectType.sol";
+import { ObjectTypes } from "../ObjectType.sol";
 import { Vec3, vec3 } from "../Vec3.sol";
 
 contract FarmingSystem is System {
-  using ObjectTypeLib for ObjectTypeId;
-
   function till(EntityId caller, Vec3 coord, uint16 toolSlot) external {
     caller.activate();
     (Vec3 callerCoord,) = caller.requireConnected(coord);
 
-    (EntityId farmland, ObjectTypeId objectTypeId) = getOrCreateEntityAt(coord);
-    require(objectTypeId == ObjectTypes.Dirt || objectTypeId == ObjectTypes.Grass, "Not dirt or grass");
+    (EntityId farmland, ObjectType objectType) = getOrCreateEntityAt(coord);
+    require(objectType == ObjectTypes.Dirt || objectType == ObjectTypes.Grass, "Not dirt or grass");
```

Comment:
@vdrg thoughts on also putting this into `ObjectType`? something like `isTillable`?

Reply:
Yeah makes sense to me